### PR TITLE
Fixed inconsistent data type, used proper envparse, removed unnecessary type casting

### LINKED ISSUE #3651: Read environment variables from file
I found that our project uses some environment variables to maintain different configuration for different deployment. But for local development, developers have to run many `export` commands. It is not convenient (we human are lazy).

My suggestion is to write down those variables in _.env_, let it ignored by Git, and use [envparse](https://github.com/rconradharris/envparse) to parse the file.

### PR DESCRIPTION
**Pull request by magdalenesuo**
#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [X] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Fixes changes proposed in [#3651(comment)](https://github.com/fossasia/open-event-orga-server/issues/3651#issuecomment-303905386)

#### Changes proposed in this pull request:

- use proper envparse feature
- remove str type cast
- fix inconsistent data type

Fixes #3651

### CODE REVIEW

**Comment by hongquan** [comment]
This is what you do:

selection_155

But this is what I told you to do:

selection_156

File: config.py [hunk 1]
```diff
@@ -109,8 +109,8 @@ class DevelopmentConfig(Config):
     SQLALCHEMY_RECORD_QUERIES = True
 
     # If Env Var `INTEGRATE_SOCKETIO` is set to 'true', then integrate SocketIO
-    socketio_integration =  env.str('INTEGRATE_SOCKETIO', default=False)
-    INTEGRATE_SOCKETIO = bool(socketio_integration == 'true')
+    socketio_integration =  env('INTEGRATE_SOCKETIO', default=False)
+    INTEGRATE_SOCKETIO = env.bool(socketio_integration == 'true')
```
Comment (magdalenesuo) [inline config.py]: Is this not where I was suppose to change? @hongquan

Comment (hongquan) [inline config.py]: Look carefully my example in my comment. Word by word.

**Comment by magdalenesuo** [comment]
@hongquan Hi, I had to revert a few data inconsistency change.
Well, yes I am aware of the inconsistent data type since you mentioned it but the logic of the code seems to work with **None** only (otherwise it breaks the whole code), because we do not want an empty string SERVER_NAME or DATABASE_URL , we either want it there or don't want it at all. 
Also, I thought env.str will make readability better. But I haven't revert that back. 
Please review again :)

**Comment by hongquan** [comment]
> but the logic of the code seems to work with None only
Ok
>  I thought env.str will make readability better
Ok

**Pull request merged**